src/hb-buffer-serialize.cc

```diff
@@ -368,7 +367,6 @@ hb_buffer_serialize_glyphs (hb_buffer_t *buffer,
 					       buf, buf_size, buf_consumed,
 					       font, flags);
 
-    default:
     case HB_BUFFER_SERIALIZE_FORMAT_INVALID:
       return 0;
```

Comment:
It says it is not needed to put default when all the cases are available on the switch.

Reply:
I don’t think removing `default` case is right here, since this is handling user input and it can be anything, so one needs to catch invalid values. What is the warning message this is fixing?

Reply:
Right.  Since this is input, we should actually cast it to int or unsigned in the switch ().  Otherwise the compiler is "within its rights" to assume default never happens.  Even with such a cast, it probably is technically still undefined-behavior if someone passed in another value.  Not sure, need to contact C/C++ lawyers.  Anyway, adding the cast means we won't get warnings for missing case items if we add new values to the enum.  I think I like the cast though.